runtime/libia2/main.c

```diff
@@ -14,6 +14,18 @@ __asm__(
 #if defined(__x86_64__)
     "pushq %rbp\n"
     "movq %rsp, %rbp\n"
+
+    // Call ia2_start making sure to preserve/restore the original arguments to main
+    "pushq %rdi\n"
+    "pushq %rsi\n"
+    "pushq %rdx\n"
+    "subq $8, %rsp\n"
+    "callq ia2_start\n"
```

Comment:
We may want to call whatever will call `ia2_set_up_tags` (once) here instead of directly calling the user-defined function.

Reply:
done (switched from calling `ia2_main` to `ia2_start`)